ntoskrnl/ke/amd64/trap.S

```diff
@@ -1213,19 +1213,18 @@ EXTERN KiSwitchKernelStack:PROC
 PUBLIC KeSwitchKernelStack
 FUNC KeSwitchKernelStack
 
+    /* Save rcx and allocate callee home space */
+    mov [rsp + P1Home], rcx
+    .savereg rcx, P1Home
     sub rsp, 40
```

Comment:
Just to understand a bit what's going on, how was this "40" byte size of stack calculated? is it like 8 * 5 because home space should contain the copy of five registers?

Reply:
Home space is 4 registers, but when we enter the function, the stack is unaligned, so we need to allocate an unaligned number of bytes to make it aligned again before calling another function.